### PR TITLE
EDSC-3817: Adding UMM-Var<>UMM-S association to be fetched in addition to UMM-Var<>UMM-C association

### PR DESCRIPTION
**Pull request by dpesall**
# Overview

### What is the feature?

EDSC fetches variable information from the UMM-Var<>UMM-S association in addition to the UMM-Var<>UMM-C association.

### What is the Solution?

- Updating GraphQL to also fetch variables from services. PR for this found [HERE](https://github.com/nasa/cmr-graphql/pull/64)
- Fetching both variable associations between collection and service to give the full list of variables on the project panel customize page.

### What areas of the application does this impact?

Project Customization Panel

# Testing

### Reproduction steps

- **Environment for testing: SIT**
- **Collection to test with: AIRS/Aqua L3 Monthly Quantization in Physical Units (AIRS-only) 5 degrees x 5 degrees V006 (AIRS3QPM) at GES DISC V006**

1 Verify that this collection has variable associations on both the collection and service level.
   1.a Note that variables "Dust_Score_A" and "Dust_Score_A_ct" are present in the collection variables but not service       variables.
2. Add this collection to your project.
3. Select "Edit Options".
4. Check "Customize".
5. Click "Edit Variables"
6. Verify that "Dust_Score_A" and "Dust_Score_A_ct" are available as customizable variable options.

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: static/src/js/util/mergeVariables.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * @param {object} variables2 - JSON object of a list of variables.
+ * @return {object} - JSON object containing the merged list variables
+ */
+export const mergeVariables = (variables1, variables2) => {
+  if (variables2.items) { if (!variables1.items) return variables2 }
+  if (!variables2.items) return variables1
+
+  const existingConceptIds = new Set(variables1.items.map((item) => item.conceptId))
+  variables2.items.forEach((item2) => {
+    if (!existingConceptIds.has(item2.conceptId)) {
```
Comment (eudoroolivares2016) [inline static/src/js/util/mergeVariables.js:14]: Add newline between variable declaration and expression here

File: static/src/js/util/mergeVariables.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+/**
+ * Returns a JSON object that is the result of two variable objects being merged
+ * and ignores items with duplicate conceptIds.
+ * @param {object} variables1 - JSON object of a list of variables.
+ * @param {object} variables2 - JSON object of a list of variables.
+ * @return {object} - JSON object containing the merged list variables
+ */
+export const mergeVariables = (variables1, variables2) => {
+  if (variables2.items) { if (!variables1.items) return variables2 }
```
Comment (eudoroolivares2016) [inline static/src/js/util/mergeVariables.js:9]:
```suggestion
  if (variables2.items) {
    if (!variables1.items) {
      return variables2
    }
  }
```

File: static/src/js/actions/project.js [hunk 3]
```diff
@@ -295,6 +296,16 @@ export const getProjectCollections = () => async (dispatch, getState) => {
                   form
                 }
               }
+              variables {
```
Comment (eudoroolivares2016) [inline static/src/js/actions/project.js:299]: You are going to get test failures from this on my PR I have to do something similar

File: static/src/js/util/mergeVariables.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  if (!variables2.items) return variables1
+
+  const existingConceptIds = new Set(variables1.items.map((item) => item.conceptId))
```
Comment (eudoroolivares2016) [inline static/src/js/util/mergeVariables.js:16]: I THINK I may be incorrect here but, https://www.geeksforgeeks.org/lodash-_-merge-method/#
might be doing the same thing if you have the correct argument on the right. We use it in `static/src/js/util/portals.js`

**Review by macrouch — CHANGES_REQUESTED** [review]
This solution isn't what the ticket is asking for. "As a user, I only want to see variables supported by my selected service" 

The ticket says there are already too many variables being displayed for a service because we are showing all variables associated to a collection. We need to be displaying only those variables associated to the service that the user selects, not every variable associated to the collection.

**Review by macrouch — APPROVED** [review]

**Review by eudoroolivares2016 — APPROVED** [review]

**Pull request merged**